google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java

```diff
@@ -462,6 +467,22 @@ private static Double convertDouble(Object o, DeserializeContext context) {
     }
   }
 
+  private static BigDecimal convertBigDecimal(Object o, DeserializeContext context) {
+    if (o instanceof Integer) {
+      return BigDecimal.valueOf(((Integer) o).intValue());
+    } else if (o instanceof Long) {
+      return BigDecimal.valueOf(((Long) o).longValue());
+    } else if (o instanceof Double) {
+      return BigDecimal.valueOf(((Double) o).doubleValue()).abs();
+    } else if (o instanceof BigDecimal) {
+      return (BigDecimal) o;
+    } else {
```

Comment:
We'll need an extra branch here to handle a String coming in.

```suggestion
    } else if (o instanceof String) {
      return new BigDecimal((String) o);
    } else {
```